src/rules/scaled-access-relationships-claim.js

```diff
@@ -0,0 +1,124 @@
+/**
+* @title Scaled Access relationship-based claims
+* @overview Adds a claim based on the relationships the subject has in Scaled Access
+* @gallery true
+* @category marketplace
+*
+* This rule adds a claim to the access token based on relationships the subject has in Scaled Access.
+* 
+* Example of a resulting claim:
+* ```
+* "https://example.com/relationships": [
+*    {
+*      "relationshipType": "is_admin_of",
+*      "to": {
+*        "id": "c7b134f9-28a2-4dcc-b345-affa18977ddf",
+*        "type": "subscription"
+*      }
+*    },
+*    {
+*      "relationshipType": "is_sports_member_of",
+*      "to": {
+*        "id": "affa18977ddf-4dcc-b345-c7b134f9-28a2",
+*        "type": "subscription"
+*      }
+*    }
+* ]
+* ```
+* This is done through an API call to Scaled Access' Relationship Management API using a machine-to-machine token.
+* More info can be found at https://docs.scaledaccess.com/?path=integration-with-auth0
+* 
+* A number of rule settings are required:
+* - SCALED_ACCESS_CLIENTID: The Client ID of the machine-to-machine application.
+* - SCALED_ACCESS_CLIENTSECRET: The Client secret of the machine-to-machine application.
+* - SCALED_ACCESS_BASEURL: The base URL for the Relationship Management API, e.g. `https://api.int.scaledaccess.com/privategroups-v2`.
+* - SCALED_ACCESS_TENANT: Your Scaled Access tenant code.
+* - SCALED_ACCESS_CUSTOMCLAIM: A namespaced custom claim name of your choice. The name must be a URL. Defaults to `https://scaledaccess.com/relationships`.
+* - SCALED_ACCESS_TOKENURL: The URL where to request the machine-to-machine token. Defaults to `https://${auth0.domain}/oauth/token`.
+* - SCALED_ACCESS_AUDIENCE: The audience in the machine-to-machine token.
+*/
+function scaledAccessAddRelationshipsClaim(user, context, callback) {
+    const fetch = require("node-fetch");
+    const { URLSearchParams } = require('url');
+    const jwt = require('jsonwebtoken');
+
+    const getM2mToken = () => {
+        if (global.scaledAccessM2mToken && global.scaledAccessM2mTokenExpiryInMillis > new Date().getTime() + 60000) {
+            return Promise.resolve(global.scaledAccessM2mToken);
+        } else {
+            const tokenUrl = configuration.SCALED_ACCESS_TOKENURL || `https://${auth0.domain}/oauth/token`;
```

Comment:
```suggestion
            const tokenUrl = configuration.SCALED_ACCESS_TOKENURL || `https://${context.request.hostname}/oauth/token`;
```

I think this will work better here. For customers using a custom domain, their token endpoint should match where the Rule logic is running. 

Thoughts @seejee?

Reply:
Also, in that case, I don't think we need a configuration value for it.

Reply:
Agreed